tests/integration/test_datasets_settings.py

```diff
@@ -75,6 +83,59 @@ def test_settings_workflow(
         configure_dataset(dataset, wrong_settings, workspace=workspace)
 
 
+@pytest.mark.parametrize(
+    "settings, workspace",
+    [
+        (TextClassificationSettings(label_schema={"A", "B"}), None),
+        (TextClassificationSettings(label_schema={"D", "E"}), "admin"),
```

Comment:
I think your tests worked since the settings were the same in both cases. I forced expecting different results